src/irmin/tree.ml

```diff
@@ -1872,40 +2030,40 @@ module Make (P : Private.S) = struct
         match proofs with
         | [] -> k acc
         | (s, p) :: rest ->
-            tree_of_proof p (fun p -> rev_elts ((s, p) :: acc) rest k)
+            tree_of_proof ~env p (fun p -> rev_elts ((s, p) :: acc) rest k)
       in
       rev_elts [] rev_proof_steps @@ fun elts ->
       let k n =
         List.iter (fun (s, elt) -> Node.add_to_findv_cache n s elt) elts;
```

Comment:
What is this addition to `findv` cache aiming? Removing doesn't seem to break the tests
https://github.com/mirage/irmin/commit/6a6f2023cfdf4bc9cc77f876949271e540f44893

Reply:
It does break the unit tests.

Reply:
Yes, this is indeed required because otherwise we just stop at the node boundary - by filling the cache we can also connect the nodes together.